packages/template-retail-react-app/app/pages/checkout/util/checkout-context.js

```diff
@@ -85,6 +87,14 @@ export const CheckoutProvider = ({children}) => {
         }
     }, [customer, basket])
 
+    /**************** Einstein ****************/
+    // Run this once when checkout begins
+    useEffect(() => {
+        if (basket && basket.productItems) {
+            einstein.sendBeginCheckout(basket)
```

Comment:
Do you know how many events einstein supports? I'm curious if the pattern we have where the client has a function for each event will scale. Maybe we need to investigate an api that looks similar to `einstein.send(einsteinEvents.beginCheckout, args)`